images/default/Dockerfile

```diff
@@ -8,7 +8,7 @@ RUN apt-get update && apt-get install -y curl gnupg2 git \
         apt-transport-https ca-certificates gnupg-agent \
         software-properties-common \
         pkg-config libffi-dev patch diffutils libssl-dev iptables kmod \
-        clang crossbuild-essential-amd64
```

Comment:
the container_test does not run in with this image. Those tests run directly on the host machine with the host filesystem and libraries (using `--TESTONLY-unsafe-nonroot`). We can't ensure that `erofs-utils` is installed on the host machines that run this test. Hmmm...

So there are 2 options:
1. Create the erofs images locally and commit it into this PR under some test_data directory and use that as a data dependency. Undo this file and remove all the logic that creates the images. Also revert changes to test/cmd/test_app.
2. Write an e2e test using this image. See test/e2e/integration_test.go. Run a container with this image and attach a bind mount to a tmp directory. In that initial container, invoke `mkfs.erofs` like you currently do and write out the image file to the bind mount. Outside the container (from the test), mount EROFS by invoking `runsc debug` with the sandbox ID. Then run the test like you do.

(1) is simpler. Lets just do that.

Reply:
I'm using `make test TARGETS=runsc/container:container_test` to run the `container_test`. With above change, that test will pass and doesn't require that `erofs-utils` is installed on the host machine. I thought it's the standard way to run this test.

Reply:
The host does require `erofs-utils`. Otherwise, the host won't have `mkfs.erofs` binary.

Reply:
IIUC, when the test is invoked by `make test TARGETS=runsc/container:container_test`, the test will run in a container launched by `make` by default:

https://github.com/google/gvisor/blob/9ff4c4593819749623b7b0ead2ca095c4c7a9e89/Makefile#L62-L63

https://github.com/google/gvisor/blob/9ff4c4593819749623b7b0ead2ca095c4c7a9e89/tools/bazel.mk#L267

https://github.com/google/gvisor/blob/9ff4c4593819749623b7b0ead2ca095c4c7a9e89/tools/bazel.mk#L192-L193

https://github.com/google/gvisor/blob/9ff4c4593819749623b7b0ead2ca095c4c7a9e89/tools/bazel.mk#L39-L40

I thought it's the standard way to run the tests because basically it won't depend on the host environment in this case. I tried `make test TARGETS=runsc/container:container_test` without `erofs-utils` installed on the host and the test can pass, and if `erofs-utils` isn't installed in above docker image, the test will fail even if `erofs-utils` is installed on the host.

I guess `DOCKER_BUILD` might be `false` in your test env. Thanks for the suggestions! I will prepare a patch using the option 1 and push a new version when it's ready.

Reply:
Ah that makes complete sense. That is why this test was passing on BuildKite and failing internally. The thing is internally we have a different build system, and so tools/bazel.mk is not used. Internally, we don't containerize the execution of bazel-like commands. The standard way of running this test is `make container-tests`, which ends up invoking `call test` too.

Sorry for the back and forth, could you revert to the older versions and just add logic to skip the test if `mkfs.erofs` binary is not present in the test environment? `t.Skipf()` should do it.

With the current approach, we have `tiny.img` files but it is not clear how they were created. Your older version was more clearer as we could see what files were being created and what options the image was being created with.

Reply:
Sure, will do. It's worth taking time to make the code better.